### PR TITLE
Added support for noDefaults as a normalize* option 

### PR DESCRIPTION
**Pull request by kunagpal**
* This prevents sensible defaults from being set when set to `true`.

### CODE REVIEW

File: lib/normalizers/v1.js [hunk 1]
```diff
@@ -17,6 +17,17 @@ var _ = require('lodash').noConflict(),
...
     },
 
+    authIdMap = {
+        awsSigV4: 'awsSigV4',
+        basic: 'basicAuth',
+        bearer: 'bearerAuth',
+        digest: 'digestAuth',
+        hawk: 'hawkAuth',
+        ntlm: 'ntlmAuth',
+        oAuth1: 'oAuth1',
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: This should be `oauth1` and `oauth2`

File: lib/normalizers/v1.js [hunk 2]
```diff
@@ -41,15 +52,26 @@ _.assign(Builders.prototype, {
      */
     auth: function (entityV1) {
         if (!entityV1) { return; }
+        // Handle the noauth/normal case for new and legacy attributes first, in that order
         if (util.notLegacy(entityV1, 'auth') && entityV1.auth) {
             return (entityV1.auth.type === 'noauth') ? { type: 'noauth' } : util.sanitizeAuthArray(entityV1);
         }
-        if (!entityV1.currentHelper) { return; }
+        if (entityV1.currentHelper === 'normal') { return { type: 'noauth' }; }
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: By reversing these two conditions, you can get rid of the `util.notLegacy` call.

File: lib/normalizers/v1.js [hunk 3]
```diff
@@ -41,15 +52,26 @@ _.assign(Builders.prototype, {
      */
     auth: function (entityV1) {
         if (!entityV1) { return; }
+        // Handle the noauth/normal case for new and legacy attributes first, in that order
         if (util.notLegacy(entityV1, 'auth') && entityV1.auth) {
             return (entityV1.auth.type === 'noauth') ? { type: 'noauth' } : util.sanitizeAuthArray(entityV1);
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: Need to add `entityV1.currentHelper = 'normal'` as well

File: lib/normalizers/v1.js (hunk 1 above)
Comment (kunagpal) [reply lib/normalizers/v1.js]: As discussed, this is the mapping from `helperAttributes.id` to `currentHelper` for v1 normalization.

File: lib/normalizers/v1.js (hunk 3 above)
Comment (kunagpal) [reply lib/normalizers/v1.js]: Done below.

File: lib/normalizers/v1.js (hunk 2 above)
Comment (kunagpal) [reply lib/normalizers/v1.js]: Will take a look. 👍

File: lib/normalizers/v1.js [hunk 4]
```diff
@@ -41,15 +52,26 @@ _.assign(Builders.prototype, {
...
-        if (!entityV1.currentHelper) { return; }
+        if (entityV1.currentHelper === 'normal') { return { type: 'noauth' }; }
+
+        var auth,
+            params,
+            mapper,
+            currentHelper = entityV1.currentHelper,
+            helperAttributes = entityV1.helperAttributes;
+
+        if (!(currentHelper || this.options.noDefaults) || (currentHelper && !helperAttributes)) { return; }
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: Can you add comments explaining the `noDefaults` option?

Comment (kunagpal) [reply lib/normalizers/v1.js]: Done. 👍

**Pull request merged**